shell/src/main/java/org/apache/kafka/shell/MetadataNodeManager.java

```diff
@@ -318,6 +320,16 @@ private void handleCommitImpl(MetadataRecordType type, ApiMessage message)
                     node.create(record.key()).setContents(record.value() + "");
                 break;
             }
+            case PRODUCER_IDS_RECORD: {
+                ProducerIdsRecord record = (ProducerIdsRecord) message;
+                DirectoryNode producerIdNode = data.root.mkdirs("lastProducerIdBlock");
+                producerIdNode.create("assignedBrokerId").setContents(record.brokerId() + "");
+                producerIdNode.create("assignedBrokerEpoch").setContents(record.brokerEpoch() + "");
+                producerIdNode.create("blockStart")
```

Comment:
@dengziming I was looking at [KIP-730](https://cwiki.apache.org/confluence/display/KAFKA/KIP-730%3A+Producer+ID+generation+in+KRaft+mode). It is true that we currently hard-code the block length, but the `AllocateProducerIds` response reserves some flexibility for a dynamic block length (i.e. the block length is returned in the response). I wonder if it would be better not to depend on the hard-coded block size. Maybe we could make this `/producerIds/nextBlockStartId` or something like that.

@mumrah What do you think?

Reply:
FYI: I opened #11747 to hopefully make the naming a little clearer.

Reply:
I split the `ProducerIdsRecord` into `lastProducerIdBlock` and `nextProducerIdBlock`. the former contains `assignedBrokerId` and `assignedBrokerEpoch`, the latter contains `firstProducerId`, this is more clear.